<?xml version="1.0" encoding="UTF-8"?>
<launch>
   <arg
   name="camera_id"
   default="0"
   />
   <arg
   name="csv_name"
   default="pointing_db.csv"
   />
   <arg
   name="serial_no"
   default=""
   />



  <include file="$(find realsense2_camera)/launch/rs_camera.launch">
    <arg name="camera" value="camera_$(arg camera_id)"/>
    <arg name="serial_no" value="$(arg serial_no)"/>
    <arg name="align_depth" value="True"/>
    <arg name="color_fps" value="15"/>
    <arg name="color_width" value="1280"/>
    <arg name="color_height" value="720"/>
    <arg name="depth_fps" value="15"/>
    <arg name="depth_width" value="1280"/>
    <arg name="depth_height" value="720"/>
  </include>

  <node pkg="tree_transform" type="database_handler_service_server.py" name="database_handler_service_server_$(arg camera_id)" respawn="true" output="screen">
    <remap from="load_srv" to="load_srv_$(arg camera_id)" />
    <remap from="upload_srv" to="upload_srv_$(arg camera_id)" />
    <remap from="update_srv" to="update_srv_$(arg camera_id)" />
    <remap from="clean_srv" to="clean_srv_$(arg camera_id)" />
    <remap from="get_poses_srv" to="get_poses_srv_$(arg camera_id)" />
  </node>
  <node pkg="tree_transform" type="transform.py" name="transform_$(arg camera_id)" respawn="true">

  </node>
  
  <node pkg="iri_finger_pointing" type="mediapipe_hands_node_db.py" name="iri_finger_pointing_$(arg camera_id)" respawn="true" output="screen">
    <param name="csv_name" value="$(arg csv_name)" /> 
    <remap from="~target_pose" to="/mediapipe_pose_camera_$(arg camera_id)" />
    <remap from="~camera/aligned_depth_to_color/image_raw" to="/camera_$(arg camera_id)/aligned_depth_to_color/image_raw" />
    <remap from="~landmarks" to="/landmarks_$(arg camera_id)" />
    <remap from="~camera/color/camera_info" to="/camera_$(arg camera_id)/color/camera_info" />
    <remap from="~tag_detections" to="/tag_detections_$(arg camera_id)" />
    <remap from="load_srv" to="load_srv_$(arg camera_id)" />
    <remap from="upload_srv" to="upload_srv_$(arg camera_id)" />
    <remap from="update_srv" to="update_srv_$(arg camera_id)" />
    <remap from="clean_srv" to="clean_srv_$(arg camera_id)" />
    <remap from="get_poses_srv" to="get_poses_srv_$(arg camera_id)" />
  </node>

  <include file="$(find iri_mediapipe_graph)/launch/node.launch">
    <arg name="node_name"             default="iri_mediapipe_graph_$(arg camera_id)"/>
    <arg name="output"                default="screen"/>
    <arg name="launch_prefix"         default=""/>
    <arg name="config_file"           default="$(find iri_mediapipe_graph)/config/params.yaml"/>
    <arg name="graph_file"            default="$(find iri_mediapipe_graph)/graphs/hand_landmark_tracking_cpu.pbtxt"/>
    <arg name="image_in_topic"        default="/camera_$(arg camera_id)/color/image_raw"/>
    <arg name="info_in_topic"         default="/camera_$(arg camera_id)/color/camera_info"/>
    <arg name="image_out_topic"       default="image_out_$(arg camera_id)/image_raw"/>
    <arg name="info_out_topic"        default="image_out_$(arg camera_id)/camera_info"/>
    <arg name="landmarks_topic"       default="landmarks_$(arg camera_id)"/>
    <arg name="landmarks_3d_topic"    default="landmarks_3d_$(arg camera_id)"/>
    <arg name="landmarks_world_topic" default="landmarks_world_$(arg camera_id)"/>
    
  </include>

  <include file="$(find apriltag_ros)/launch/continuous_detection.launch">
	<arg name="node_namespace" default="apriltag_ros_continuous_node_$(arg camera_id)" />
	<arg name="camera_name" default="/camera_$(arg camera_id)/color" />
	<arg name="image_topic" default="image_raw" />
        <arg name="camera_id" default="$(arg camera_id)" />
        

  </include>
</launch>
